source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -2505,187 +2515,206 @@ void Graph::addLink(ed::PinId inputPinId, ed::PinId outputPinId)
         return;
     }
 
-    if (inputPin->_connected == false)
+    // Perform kind check
+    bool kindsMatch = (outputPin->_kind == inputPin->_kind);
+    if (kindsMatch)
     {
-        int upNode = getNodeId(inputPinId);
```

Comment:
Aside: This diff looked much better on my computer. Sorry its come across this way.